apps/src/sites/studio/pages/levels-lablab-main.js

```diff
@@ -0,0 +1,60 @@
+import $ from 'jquery';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {Provider, useSelector} from 'react-redux';
+import {getStore} from '@cdo/apps/redux';
+import LabContainer from '@cdo/apps/code-studio/components/LabContainer';
+import {levelsForLessonId} from '@cdo/apps/code-studio/progressReduxSelectors';
+import {getStandaloneProjectId} from '@cdo/apps/labs/projects/utils';
+import ProjectContainer from '@cdo/apps/labs/projects/ProjectContainer';
+import ProgressContainer from '@cdo/apps/labs/progress/ProgressContainer';
+import StandaloneVideo2 from '@cdo/apps/standaloneVideo2/StandaloneVideo2';
+import MusicView from '@cdo/apps/music/views/MusicView';
+import {logError} from '@cdo/apps/music/utils/MusicMetrics';
+
+$(document).ready(function () {
+  ReactDOM.render(
+    <Provider store={getStore()}>
+      <LabContainer
+        onError={(error, componentStack) =>
+          logError({error: error.toString(), componentStack})
+        }
+      >
+        <LabLab />
+      </LabContainer>
+    </Provider>,
+
+    document.getElementById('lablab-container')
+  );
+});
+
+const LabLab = () => {
+  const currentApp = useSelector(
```

Comment:
Can we get this from the backend directly? i feel a little wary of relying on progress to determine the app type - i'm not sure in what situation we'd be switching levels without progress, but feels like it would be better to decouple ourselves given that we're already retrieving other level data from the backend?

Reply:
It's a good point.  The challenge is that we would like to have this information for all levels available up front so that we can start each switch instantly, and then the per-level level data is retrieved.  So far, we've been using `progressRedux` as our source of truth for the set of levels, so it feels like a natural fit to also store the app type there.  The bigger change would be to factor information for the available set of levels into a more general redux store.